### DISCUSSION TITLE
Authentication LinkedIn "Server Error Unable to exchange external code"

### DISCUSSION THREAD

**Discussion by daniel-meyhoefer** [supabase/supabase#7695] [Questions]
Hello Supabase-community ✨

So far I really love supabase, especially of course that it is open source, the great docs and UX  🥰

**Note:**  I triple checked client id, client pwd and the callback_uri, all three are a-ok. I even created an additional LinkedIn-App to test for somehow faulty credentials, without any luck. So unfortunately the old thread from the faq does not really help here...  #1192  What could be other causes for this issue?

Trying to create a small website with Svelte and supabase. Wanting to use LinkedIn auth and followed the instructions (inserting project url in linkedIn App and LinkedIn credentials in supabase). 
I can click on the LogIn button, type in my LinkedIn-credentials, allow my website to access my account and get redirected. However instead of a successful session I get the following error message in my URL bar 
`error=server_error&error_description=Unable+to+exchange+external+code%3A+AQSHg_omRSOVd33nWtE8o38xuU...`
LogIn via magic link does work however... 

Google search did not really reveal anything about that error... I used for my Svelte project the Supabase documentation skeleton
([https://supabase.com/docs/guides/with-sveltekit](url))

Many thanks in advance for help! Right now, I do not even know, where exactly I have to check for errors...

**Edit:** Tested the project with a different provider than LinkedIn (i.e. Discord) worked like a charm. So possible that there is a bug either on supabase or LinkedIn side?

**Comment by ccll** [comment]
Same error with sveltekit + supabase + auth-helper-svelte + Google Auth.
The client id and secret are correct without hidden whitespaces.

**Comment by Matangub** [comment]
I'm encountering the same bug. have you maybe managed to solve it? @daniel-meyhoefer @ccll

**Comment by taras-yakymiuk** [comment]
I got this error when I'm trying to login with an Linkedin account that does not have avatar (profile img). You can test it out by adding a profile img (in case it's missing). Anyways that's not a workaround since it's ok to not have an avatar

**Reply by matanAlltra** [reply]
thank you that really was the solution. New accounts that don't have a profile image will fail to authenticate. @taras-yakymiuk

**Reply by taras-yakymiuk** [reply]
@matanAlltra I created an issue for that
https://github.com/supabase/gotrue-js/issues/488
this is sort of not ok since lots of accounts have no picture
you can upvote this issue, maybe so it will be resolved sooner

**Comment by lsrugo** [comment]
I am having the same issue. Adding a profile photo solved it. @taras-yakymiuk were you able to find a fix?
